asset/asset.go

```diff
@@ -487,6 +487,79 @@ func (g *GroupKey) IsEqualGroup(otherGroupKey *GroupKey) bool {
 	return g.GroupPubKey.IsEqual(&otherGroupKey.GroupPubKey)
 }
 
+// IsGroupSig attempts to parse the given witness as a single Schnorr signature,
+// and returns the signature and true if it is valid.
+func IsGroupSig(witness wire.TxWitness) (*schnorr.Signature, bool) {
+	if len(witness) != 1 {
```

Comment:
These functions also need to mind the annex if we're going for 1:1 equivalence. There's some functions in `btcd/txscript` that should be useful for that.

Reply:
Or see https://github.com/lightninglabs/pool/blob/master/poolscript/script.go#L466

Reply:
Totally forgot about the annex, thanks for the catch - So IIUC for the current sig-only functionality, we should now allow any annex as long the first element is a sig?